### PR TITLE
Add FlowsClient.delete_flow

### PR DESCRIPTION
**Pull request by sirosen**
- new method
- new test fixture data
- single new test case

### CODE REVIEW

File: src/globus_sdk/_testing/data/flows/delete_flow.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    "visible_to": [],
+}
+
+RESPONSES = ResponseSet(
+    metadata={"flow_id": TWO_HOP_TRANSFER_FLOW_ID},
+    default=RegisteredResponse(
+        service="flows",
+        path=f"/flows/{TWO_HOP_TRANSFER_FLOW_ID}",
+        method="DELETE",
+        json=TWO_HOP_TRANSFER_FLOW_DOC,
```
Comment (kurtmckee) [inline src/globus_sdk/_testing/data/flows/delete_flow.py:41]: Should this be `_DELETED_DOC`?

Comment (sirosen) [reply src/globus_sdk/_testing/data/flows/delete_flow.py]: _amend and force push_
Isn't that what I said?

File: src/globus_sdk/services/flows/client.py [hunk 2]
```diff
@@ -237,6 +237,27 @@ def list_flows(
 
         return IterableFlowsResponse(self.get("/flows", query_params=query_params))
 
+    @_flowdoc("List Flows", "Flows/paths/~1flows~1{flow_id}/delete")
```
[the remark this answers is not in the document]
Comment (sirosen) [reply src/globus_sdk/services/flows/client.py]: Thank you for catching this. As is perhaps obvious, I did this with a copy-paste-replace and I missed something here.

File: tests/functional/flows/test_delete_flow.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+from globus_sdk import FlowsClient
+from globus_sdk._testing import load_response
+
+
+def test_delete_flow(flows_client: FlowsClient):
+    metadata = load_response(flows_client.delete_flow).metadata
+
+    resp = flows_client.delete_flow(metadata["flow_id"])
+    assert resp.data["title"] == "Multi Step Transfer"
+    assert resp.data["DELETED"] is True
```
Comment (kurtmckee) [inline tests/functional/flows/test_delete_flow.py:10]: Nice. :grinning:

**Review by kurtmckee — APPROVED** [review]

**Pull request merged**